File: api/src/main/java/fr/lule/acad/aggregate/Journal.java
package fr.lule.acad.aggregate;

import com.google.common.eventbus.EventBus;

import fr.lule.acad.event.JournalDay;
import fr.lule.acad.event.JournalEntryEdited;
import fr.lule.acad.util.IDateFactory;

public class Journal {

	public static boolean editJournalEntry(EventBus bus, String day, String text, IDateFactory dateFactory) {
		bus.post(new JournalEntryEdited(text, new JournalDay(day), dateFactory.now()));
		return true;
	}

}

File: app/src/reducers/journal.js
const journal = (state = [], action) => {
    switch (action.type) {
        case "FETCH_JOURNAL_SUCCESS":
            return action.journal;
        case "EDIT_JOURNAL_ENTRY":
            return state.map(entry =>
                (entry.day === action.day)
                    ? { ...entry, text: action.newText }
                    : entry
            )
        default:
            return state
    }
}

export default journal
